File: cco/cco15p4/cco15p4.py
def dfs(x, y):
	if grid[y][x] > "V":
		[dfs(xx, y) for xx in xrange(x - 1, -1, -1) if grid[y][xx] > "."]
	elif grid[y][x] > "R":
		[dfs(x, yy) for yy in xrange(y + 1, gy) if grid[yy][x] > "."]
	elif grid[y][x] > "M":
		[dfs(x, yy) for yy in xrange(y - 1, -1, -1) if grid[yy][x] > "."]
	elif grid[y][x] > "D":
		[dfs(xx, y) for xx in xrange(x + 1, gx) if grid[y][xx] > "."]
	grid[y][x] = "."
	print("({},{})".format(y, x))


input = __import__("sys").stdin.readline
gy, gx = map(int, input().split())
grid = [list(input()) for _ in xrange(gy)]
for y in xrange(gy):
	for x in xrange(gx):
		if grid[y][x] != ".":
			dfs(x, y)

File: cco/cco15p4/cco15p4.cpp
#include <iostream>

char grid[2000][2001];
int gx, gy;

void dfs(int x, int y) {
	if (grid[y][x] < 70) {
		for (int xx = x + 1; xx < gx; xx++) {
			if (grid[y][xx] > 46) {
				dfs(xx, y);
			}
		}
	} else if (grid[y][x] < 79) {
		for (int yy = y - 1; yy >= 0; yy--) {
			if (grid[yy][x] > 46) {
				dfs(x, yy);
			}
		}
	} else if (grid[y][x] < 84) {
		for (int yy = y + 1; yy < gy; yy++) {
			if (grid[yy][x] > 46) {
				dfs(x, yy);
			}
		}
	} else if (grid[y][x] < 88) {
		for (int xx = x - 1; xx >= 0; xx--) {
			if (grid[y][xx] > 46) {
				dfs(xx, y);
			}
		}
	}
	grid[y][x] = 46;
	printf("(%d,%d)\n", y, x);
}

int main() {
	scanf("%d %d", &gy, &gx);
	for (int y = 0; y < gy; y++) {
		scanf("%s", grid[y]);
	}
	for (int y = 0; y < gy; y++) {
		for (int x = 0; x < gx; x++) {
			if (grid[y][x] > 46) {
				dfs(x, y);
			}
		}
	}
	return 0;
}